### ISSUE TITLE
Collaboration proposition

### ISSUE THREAD

**Issue by me-no-dev** [flyduino/kissfc-wifi-module#2]
As one of the ESP8266-Arduino core developers, and particularly responsible for much of the networking, I would like to offer you my "help" into better integrating kiss and wifi.
The reason that I file an issue and not jus PR something is that the changes would be massive and are something worth discussing first.
Let me know how you feel about it and we can discuss it further.

**Comment by fedorcomander** [comment]
sure thing. its my first try with esp. may be not super pretty. what do u have in mind? 

---
Cheers,
Alex

**Comment by me-no-dev** [comment]
I already have something similar in terms of network (https://github.com/me-no-dev/esp8266fcgui/blob/master/esp8266fcgui.ino). It uses Asynchronous networking for both HTTP and WebSocket. At the time I wrote it, it was meant to be a simple WS to Serial firmware with KISS and BF GUIs onboard. The sketch though also has DNS (captive portal on AP), SSDP and NetBIOS presence (for those Windows users).
On top of that, MDNS can be better integrated and then discovered from the WebGui (I checked that Chrome has the API for mDNS) so you boot the quad and select it from the dropdown :).
So if a mix of two can be made, I think a much more powerful device can be achieved. (I can write the serial driver to follow Kiss protocol as well).

**Comment by me-no-dev** [comment]
I forgot to mention that the OTA protocol can also be implemented into the Chrome GUI so you can update the ESP from the browser.

**Comment by fedorcomander** [comment]
Yeah, i saw your code. I DIDNT put GUI on the device on purpose, actually. Just don't want people to update one more thing with every gui update :) Sure with OTA is a breathe, but still, trying to keep it KISSimple ;) So far it works ok. I do have captivity issue (phone home) on ios 10.x, will fix it later this week, a bit busy now with other things. Just need to intecept some urls and respond with OK... Already did it for some captive portal i was coding like 2 years back.

I didn't really want to integrate betaflight, because its KISS WIFI after all ;)... Also, i do parse packets and stuff, so it is very much kiss related. 

GUI must be working on ALL devices (win/linux/osx/ios/android), i am not sure how underlying browser implementation (cordova, nodejs) will support mdns resolving... So there is more then just chrome ;)))

What i see, you are doing pretty much the same thing (WS->SERIAL)... One exception that i am using "native" clients in place of browser... 

Also, can give you small advice... Kiss can be working in different baud rates (115200 or 250000)... Which you do not support atm. May be a good addition. You need to auto detect it btw.

I have 2 host names, actually... kiss.ws and kiss.local (one for dns and one for mdns)... GUI tries to connect to both of them and successful connect wins ;) May be its not super "right" solution but looked ok at a time.

Feel free to fork, or do whatever you think with the code, its GPL anyways.

Cheers
Alex

**Comment by me-no-dev** [comment]
Well I have no desire to keep that project running ;) It was just a test and I needed it at the time as iPhone user and no iOS app at the time. I don't think you understood what I meant by pointing at that sketch. I wanted to make things a bit better and that be used, rather that opening yet another repository in my account. Regardless of what browser is used, when we talk about Chrome GUI, then it's only Chrome so, mDNS can be added and device can be discovered and added to the dropdown like any other serial. I do not mind writing that code either. I did not mean to add onboard GUI, or at least not FC GUI. Maybe WiFi settings ;) I am well aware of the issues to have to update the file system ;)

**Comment by fedorcomander** [comment]
Well, i dont thing MASSIVE change is really needed... If you think you can (i am sure u can) improve things, go ahead and make PR... :) Any changes which makes things better are always welcome opf course.

**Comment by me-no-dev** [comment]
By Massive I probably meant changes that are far bigger than usual PRs. Requiring another library (or two in this case) is not for underestimating either :)

**Comment by fedorcomander** [comment]
pr will do the trick. 

---
Cheers,
Alex
